Formula/openssl110.rb

```diff
@@ -0,0 +1,184 @@
+class Openssl110 < Formula
+  desc "Cryptography and SSL/TLS Toolkit"
+  homepage "https://openssl.org/"
+  url "https://www.openssl.org/source/openssl-1.1.0-pre5.tar.gz"
+  mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.1.0-pre5.tar.gz"
+  version "1.1.0-beta5"
+  sha256 "25acbdfa5e0259ed20159670e88ddb4257970f80ce923427bd201133e6e580db"
+
+  keg_only :provided_by_osx,
+    "Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries"
+
+  option :universal
+  option "without-test", "Skip build-time tests (not recommended)"
+
+  # Only needs 5.10 to run, but needs >5.13.4 to run the testsuite.
+  # https://github.com/openssl/openssl/blob/4b16fa791d3ad8/README.PERL
+  if build.with? "test"
+    depends_on :perl => "5.14"
+  else
+    depends_on :perl => "5.10"
+  end
```

Comment:
> Because this is the only safe assumption, as the bottle may hardcode link to the default_formula(one of examples is this openssl formula itself).

We'll catch that with `brew linkage` now, though.